libs/s25main/buildings/nobBaseWarehouse.h

```diff
@@ -247,7 +247,7 @@ class nobBaseWarehouse : public nobBaseMilitary, public DataChangedObservable
                + GetNumRealFigures(Job::Sergeant) + GetNumRealFigures(Job::Officer) + GetNumRealFigures(Job::General);
     }
     /// Bestellt Soldaten
-    void OrderTroops(nobMilitary* goal, unsigned count, bool ignoresettingsendweakfirst = false);
+    void OrderTroops(nobMilitary* goal, std::array<unsigned, NUM_SOLDIER_RANKS> &counts, unsigned &max);
```

Comment:
Please provide a docstring what this is supposed to do especially in regard to the (IMO unusual) reference parameters